service.py

```diff
@@ -144,12 +130,14 @@ def get_biospecimen_kf_id(self, external_sample_id, study_id):
         while retry_count > 1:
             resp = requests.get(
                 self.api+'/biospecimens?study_id='+study_id +
-                '&external_sample_id='+external_sample_id)
+                '&external_sample_id='+external_sample_id,
+                timeout=self.context.get_remaining_time_in_millis()-13000)
             if resp.status_code != 500:
                 break
             else:
-                time.sleep(1)
                 retry_count = retry_count - 1
+        if resp.status_code != 200:
+            raise TimeoutException
         if resp.status_code == 200 and len(resp.json()['results']) == 1:
```

Comment:
if resp.status_code == 200 not needed here now because it raises an exception otherwise